Click outside

OksanaSam opened the pull request:
<!-- This PR template was borrowed from the fine folks at The Collab Lab/Zapier https://the-collab-lab.codes/ -->

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
- Added click outside hook for the dropdown menu to disappear on click outside;
- Swap index.html and Home link in Home route;
- Add the correct link to Events routing 

<!-- What does this code change? Why did I choose this approach? Did I learn anything worth sharing? -->

## Related Issue

<!-- Link Jira ticket here using this Markdown format: (Link name)[URL] -->

## Acceptance Criteria

<!-- Include the requirements from your Jira ticket -->

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

<!-- If UI feature, provide screenshots or gif https://ezgif.com/video-to-gif -->

### After
<img width="691" alt="Screen Shot 2021-02-07 at 10 53 19 PM" src="https://user-images.githubusercontent.com/60159629/107174229-abe7b780-6997-11eb-9f76-55c6faf6bb74.png">

<!-- If UI feature, provide screenshots or gif https://ezgif.com/video-to-gif -->

## Testing Steps / QA Criteria

<!-- Provide steps the code reviewer needs to follow to properly test your additions. -->
- Try clicking on all routes (provided they are active) and clicking outside to check if the routing is working

neilhsieh commented:
This looks good so far, got 2 minor things to take a look at:

1. For the `useOnClickOutside` you could also remove the callback function from the dependency array so its just [ref], minor change, but this way, its only being called once, and not again when the function is run.
2. For the HeaderMenu component, the console is throw us warnings about  DOMNesting, and that's happening since we have a tags nested in a tags, that happens in React where there is a <Link> inside of the <a> tag (since the Link renders to an HTML a tag anyways).

OksanaSam commented:
@neilhsieh Thanks a lot for the feedback! Fixed both issues. I also deleted "coming soon" <a> tag in About Link as it was causing the problem. We will have About page ready soon so won't need it anyway. Please check if the fixes look good to you.
